conda_forge_tick/git_utils.xsh

```diff
@@ -226,22 +240,30 @@ def push_repo(feedstock_dir, body, repo, title, head, branch,
         # Copyright (c) 2016 Aaron Meurer, Gil Forsyth
         token = $PASSWORD
         deploy_repo = $USERNAME + '/' + $PROJECT + '-feedstock'
-        doctr_run(['git', 'remote', 'add', 'regro_remote',
-                   'https://{token}@github.com/{deploy_repo}.git'.format(
-                       token=token, deploy_repo=deploy_repo)],
-                  token=token.encode('utf-8'))
-
-        doctr_run(['git', 'push', '--set-upstream', 'regro_remote', branch],
-                  token=token.encode('utf-8'))
+        if dry_run:
+            repo_url = 'https://github.com/{deploy_repo}.git'.format(deploy_repo=deploy_repo)
+            print("dry run: adding remote and pushing up branch for %s" % repo_url)
+        else:
+            doctr_run(['git', 'remote', 'add', 'regro_remote',
+                       'https://{token}@github.com/{deploy_repo}.git'.format(
+                           token=token, deploy_repo=deploy_repo)],
+                      token=token.encode('utf-8'))
+
+            doctr_run(['git', 'push', '--set-upstream', 'regro_remote', branch],
+                      token=token.encode('utf-8'))
     # lastly make a PR for the feedstock
     if not pull_request:
         return
     print('Creating conda-forge feedstock pull request...')
-    pr = repo.create_pull(title, 'master', head, body=body)
-    if pr is None:
-        print('Failed to create pull request!')
+    if dry_run:
+        print("dry run: create pr with title: %s" % title)
+        return False
     else:
-        print('Pull request created at ' + pr.html_url)
+        pr = repo.create_pull(title, 'master', head, body=body)
+        if pr is None:
+            print('Failed to create pull request!')
```

Comment:
Not that this is necessarily in scope for this PR but we might consider returning false here, since we're running inside a try except and could be a bit more graceful.